django_webapps/scripts/db_population/populate_teachers.py

```diff
@@ -0,0 +1,224 @@
+    "name": 4,
+    "email": 5,
+    "tg": 6,
+    "timezone": 7,
+    "experience": 8,
+    "groups_number": 12,
+    "age_ranges": 13,
+    "language_levels": 14,
+    "mon": 15,
+    "tue": 16,
+    "wed": 17,
+    "thu": 18,
+    "fri": 19,
+    "sat": 20,
+    "sun": 21,
+    "speaking_club": 22,
+    "can_give_feedback": 23,
+    "other_help_comment": 24,
+}
+
+
+def warning_message(column_name: str, value: str) -> str:
+    return f"{column_name} is not provided or can not be parsed, value: {value}"
+
+
+def get_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input_csv", "-i", required=True, help="Input csv file with teachers")
+    return parser.parse_args()
+
+
+def load_teachers(path_to_csv_file: str) -> list[list[str]]:
+    with open(path_to_csv_file, newline="") as csvfile:  # noqa: PTH123
+        return list(csv.reader(csvfile))
+
+
+class TeacherPopulator:
+    def __init__(self, teachers: list[list[str]]) -> None:
+        self.teachers = teachers[1:]
+        self.header = teachers[0]
+        self._current_teacher = None
+
+    def run(self) -> None:
+        for teacher in tqdm(self.teachers, desc="Going through the tachers..."):
+            tid = teacher[COLUMN_TO_ID["tid"]]
+            logger.info("======================================================= ")
+            logger.info(f"Parsing teacher with tid {tid}")
+            self._current_teacher = teacher
+            self._create_teacher()
+
+    @transaction.atomic
+    def _create_teacher(self) -> None:
+        if self._current_teacher is None:
+            raise ValueError("Current teacher can not be None to call")
+
+        status = self._parse_column("status", teacher_parsers.parse_status)
+
+        tid = int(self._current_teacher[COLUMN_TO_ID["tid"]])
+
+        if (
+            status is not None
+            and status[0] == TeacherSituationalStatus.NO_RESPONSE
+            and tid > MAX_TID_WITH_NO_RESPONSE
+        ):
+            logger.info("Skipping teacher since no response from him")
+            return
+        none_teaching_help_comment = self._current_teacher[COLUMN_TO_ID["other_help_comment"]]
+        has_experience = self._parse_column("experience", teacher_parsers.parse_has_experience)
+        can_give_feedback = self._parse_column(
+            "can_give_feedback", teacher_parsers.parse_can_give_feedback
+        )
+        groups_number = self._parse_column("groups_number", teacher_parsers.parse_groups_number)
+        speaking_club = self._parse_column("speaking_club", teacher_parsers.parse_speaking_club)
+        personal_info = self._create_personal_info()
+        availability_slots = self._create_availability_slots()
+        languages_and_levels = self._create_language_and_levels()
+        age_ranges = self._create_age_ranges()
+        try:
+            teacher = Teacher.objects.create(
+                project_status=status[0] if type(status[0]) == TeacherProjectStatus else None,
```

Comment:
Probably clarification is needed. What if we have only situational status and not project status, can this happen? If we can't we we might need to include project status for all situational statuses since they all are mixed

Reply:
btw, project_status can not be null so if we have situational status we have to include project status